### PR TITLE
Fix input date issues

### LINKED ISSUE #1357: [InputDate] onChange is not called when you empty the value
`<InputDate       onChange={this._onChangeDateRatt}
                            value={dateRattForm}
                            error={errorRatt}
                            format={format}
                            label={translate('contrat.dateRattachement')}
/>`
# Current behaviour
if I chose a value, _onChangeDateRatt is called but not when i empty the value.

# Expected behaviour
_onChangeDateRatt should be called whenever the value changes

## Versions

Focus-core: 2.1.1
Focus-components: 2.1.4

### PR DESCRIPTION
**Pull request by Hartorn**
Fix #1357 and #1325 

Still have to test in details before merging

### CODE REVIEW

File: src/components/input/date/index.js [hunk 1]
```diff
@@ -147,23 +149,17 @@ class InputDate extends Component {
...
-            if (isCorrect) {
-                this.props.onChange(dropDownDate.toISOString());
-            }
-            return;
-        }
-        if (fromBlur !== true && isCorrect) {
-            this.props.onChange(dropDownDate.toISOString());
+        // Fire onChange event, only if date if correct, or empty, if the option is activated
+        if (fromBlur !== true && (isCorrect || (triggerOnChangeIfEmpty && (inputDate || '').trim() === ''))) {
+            this.props.onChange(isCorrect ? dropDownDate.toISOString() : null); // if date is not correct, it is empty, so send null (or empty string ?)
```
Comment (fconstantin) [inline src/components/input/date/index.js]: Not more condition on checkOnlyOnBlur ?

**Comment by Hartorn** [comment]
@fconstantin Oupsie, was tired I think. Did not see the second part of the condition

**Comment by Hartorn** [comment]
@fconstantin I think I should add also the option in accordance with both check

**Comment by Hartorn** [comment]
@fconstantin Should be good

**Pull request merged**